sunpy/database/tests/test_database.py

```diff
@@ -840,3 +840,44 @@ def test_disable_undo(database, download_query, tmpdir):
         db.clear()
     with pytest.raises(EmptyCommandStackError):
         database.undo()
+
+
+def test_split_function(filled_database, database):
+    i=1
+    for entry in filled_database:
+        # every fourth entry gets the instrument 'EIA'
+        if i % 4 == 0:
+            filled_database.edit(entry, instrument='EIA')
+            filled_database.remove_tag(entry, 'foo')
+        # every fifth entry gets the instrument 'AIA_3'
+        elif i % 5 == 0:
+            filled_database.edit(entry, instrument='AIA_3')
+            filled_database.remove_tag(entry, 'bar')
+        #every other entry gets instrument 'RHESSI'
+        else:
+            filled_database.edit(entry, instrument='RHESSI')
+        #all  entries have provider 'xyz'
+        filled_database.edit(entry, provider='xyz')
+        i=i+1
+    filled_database.commit()
```

Comment:
Write a custom fixture for this edited database. This way there will be a cleaner separation between creating data and checking equalities.